frontend/app/model/RichEvent.scala

```diff
@@ -1,14 +1,39 @@
 package model
 
+import com.github.nscala_time.time.Imports._
 import com.gu.contentapi.client.model.Content
 import configuration.Links
 import controllers.routes
 import model.EventMetadata.{HighlightsMetadata, Metadata}
 import model.Eventbrite.EBEvent
+import org.joda.time.LocalDate
 import services.MasterclassData
 
+import scala.collection.immutable.SortedMap
+
 object RichEvent {
 
+  def chronologicalSort(events: Seq[RichEvent]) = {
+    events.sortWith(_.event.start < _.event.start)
+  }
+
+  def groupEventsByDay(events: Seq[RichEvent])(implicit chronologicalOrdering: Ordering[LocalDate]): SortedMap[LocalDate, Seq[RichEvent]] = {
+    val unsortedMap = events.groupBy(_.start.toLocalDate)
+    SortedMap(unsortedMap.toSeq :_*)(chronologicalOrdering)
+  }
+
+  def groupEventsByDayAndMonth(events: Seq[RichEvent])(implicit chronologicalOrdering: Ordering[LocalDate]):  SortedMap[LocalDate, SortedMap[LocalDate, Seq[RichEvent]]] = {
+    val unsortedMap = groupEventsByDay(events)(chronologicalOrdering).groupBy(_._1.withDayOfMonth(1))
+    SortedMap(unsortedMap.toSeq :_*)(chronologicalOrdering)
+  }
+
+  def groupEventsByMonth(events: Seq[RichEvent]): SortedMap[LocalDate, Seq[RichEvent]] = {
+    val unsortedMap = events.groupBy(_.start.toLocalDate.withDayOfMonth(1)).map {
```

Comment:
You should be able to lose the call to `map` and inline everything into one expression

```SortedMap(events.groupBy(_.start.toLocalDate.withDayOfMonth(1)).toSeq :_*)```